projects/bb3/agents/r2c2_bb3_agent.py

```diff
@@ -421,6 +421,7 @@ def __init__(self, opt, shared=None):
             self.agents[Module.SEARCH_KNOWLEDGE] = agent
 
         self.memories = []
+        self.in_session_memories = set()
```

Comment:
Do we need to have this as a set? Why not relying on the memory deduplication algorithm we had before?

Reply:
the `set` is just for quick O(1) checking in the `get_available_memory` function for memory utils

memory in list --> O(n)
memory in set --> O(1)

Reply:
OK, I see. I thought you had that for deduplication. That makes sense.